src/vs/base/browser/ui/list/listWidget.ts

```diff
@@ -892,7 +892,7 @@ export class DefaultStyleController implements IStyleController {
 		}
 
 		if (styles.listActiveSelectionIconForeground) {
-			content.push(`.monaco-list${suffix}:focus .monaco-list-row.selected .codicon { color: ${styles.listActiveSelectionIconForeground}; }`);
+			content.push(`@layer monaco-list { .monaco-list${suffix}:focus .monaco-list-row.selected .codicon { color: ${styles.listActiveSelectionIconForeground}; } }`);
```

Comment:
I think these rules are overly generic as they end up theming every codicon that appears anywhere in a chat item

Long term I think it would be better to move this css to the places that actually need it, such as the suggest widget. Adding the `@layer` is a hack that lets the chat item css basically invalidate these rules. I could not find a better way to do this without duplicating a whole lot of codicon coloring css

Reply:
I don't understand what this does, how does it change how the rule is applied?

Reply:
It lets us use `revert-layer` value in the anchor css to clear this style rule. I couldn't figure out any other way of doing this

Reply:
That would be much better actually. Tried out but not working with just setting those properties to undefined/empty string . Seeing if there's a simple fix to get it working

Reply:
Ugh looks like we end up with the same issue actually. The problem is that the override would needs to invalidate the global `.monaco-list .codicon` rule. Simply having the override use `unset` or not set any color causes the global rule to be applied instead

The `@layer` trick seems like the only way to do this

Reply:
I have another issue open for the global list rules, which I don't think should exist. I often override them with `!important` but they cause issues in many lists for codicons. The `@layer` trick is fine for now.